instrumentation/internal/internal-reflection/bootstrap/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/reflection/RealInterfaces.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.internal.reflection;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+public final class RealInterfaces {
+  private static final Map<Class<?>, Class<?>[]> realInterfaces =
+      Collections.synchronizedMap(new WeakHashMap<>());
+
+  private RealInterfaces() {}
+
+  /**
+   * Returns the interfaces directly implemented by the given class or interface. We instrument
+   * Class#getInterfaces() to remove interfaces added by us. This method can be used to get the
+   * interfaces that the class directly implements including the interfaces that we have added.
+   *
+   * @param clazz class or interface
+   * @return interfaces directly implemented by given class
+   */
+  public static Class<?>[] get(Class<?> clazz) {
+    // instrumentation calls RealInterfaces#set() from Class#getInterfaces()
```

Comment:
oh got it, ya, how about
```suggestion
    // clazz.getInterfaces() needs to be called before reading from realInterfaces
    // as the call to clazz.getInterfaces() trigger adding clazz to that map
    // via instrumentation calling RealInterfaces#set() from Class#getInterfaces()
```